public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx

```diff
@@ -39,6 +39,28 @@ export const MegaMenu = memo(
       .filter((item) => item.id !== 'profile' && item.id !== 'help')
       .map((item) => enrichWithInteractionTracking(item, state.megaMenuDocked));
 
+    if (config.featureToggles.pinNavItems) {
+      const bookmarksItem = findByUrl(navItems, '/bookmarks');
+      if (bookmarksItem) {
+        // Add children to the bookmarks section
+        bookmarksItem.children = pinnedItems.reduce((acc: NavModelItem[], url) => {
+          const item = findByUrl(navItems, url);
+          if (!item) {
+            return acc;
+          }
+          acc.push({
+            ...item,
+            children: undefined,
+            sortWeight: 0,
+            emptyMessageId: '',
+            emptyMessage: '',
```

Comment:
do we need these? i think they're all optional props anyway

Reply:
We are clearing these props for the ones that do have it, otherwise in the bookmarks section these things appear. Like if you bookmark "Starred":
<img width="297" alt="Screenshot 2024-08-15 at 11 25 45" src="https://github.com/user-attachments/assets/087ea4bc-e1b1-4cbf-9816-7e5cf1833cb5">

Reply:
ahhhh gotcha 👍 i wonder then if it's safer to only push an item with a specified list of props, rather than spreading the whole item and clearing props 🤔 
e.g.
```tsx
  acc.push({
    id: item.id,
    text: item.text,
    url: item.url,
    parentItem: { id: 'bookmarks', text: 'Bookmarks' }
  })
```
in the reduce function instead? that way there's no chance of us forgetting to clear some new prop we might add

idk, happy for you to merge either way 🙇

Reply:
That's a good point, I've done it this way mostly out of not knowing what all the "important" props are 😅  For instance, what do `highlightText` and `highlightId` do and are they also important to display the nav text?

I can change it to that and see if anything is broken in the items I can test!

Reply:
> instance, what do highlightText and highlightId do and are they also important to display the nav text?

Those are for the `feature highlights` feature, which afaik we're not using anymore.

Reply:
I'll push this change then and then we can test it during the rollout phase to see if any nav items are missing any important prop 👍